File: src/test/java/nz/co/searchwellington/commentfeeds/detectors/BlogspotCommentFeedDetectorTest.java
package nz.co.searchwellington.commentfeeds.detectors;

import junit.framework.TestCase;

import java.net.MalformedURLException;
import java.net.URL;

public class BlogspotCommentFeedDetectorTest extends TestCase {

    final static URL BLOGSPOT_COMMENT_URL = urlOf("http://wellurban.blogspot.com/feeds/113750684886641660/comments/default");
    final static URL BLOGSPOT_ATOM_URL = urlOf("http://wellurban.blogspot.com/atom.xml");

    CommentFeedDetector detector = new BlogspotCommentFeedDetector();
    
    
    public void testShouldDetectBlogspotCommentFeedUrl() {
        assertTrue(detector.isValid(BLOGSPOT_COMMENT_URL));
    }
    
    
    public void testShouldIgnoreBlogspotAtomFeed() {
        assertFalse(detector.isValid(BLOGSPOT_ATOM_URL));
    }

    private static URL urlOf(String url) {
        try {
            return new URL(url);
        } catch (MalformedURLException e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/test/scala/nz/co/searchwellington/commentfeeds/detectors/BlogspotCommentFeedDetectorTest.scala
package nz.co.searchwellington.commentfeeds.detectors

import nz.co.searchwellington.model.Newsitem
import org.junit.jupiter.api.Assertions.{assertFalse, assertTrue}
import org.junit.jupiter.api.Test

import java.net.URL

class BlogspotCommentFeedDetectorTest {

  private val BLOGSPOT_COMMENT_URL = new URL("http://wellurban.blogspot.com/feeds/113750684886641660/comments/default")
  private val BLOGSPOT_ATOM_URL = new URL("http://wellurban.blogspot.com/atom.xml")
  private val source = Newsitem()

  private val detector = new BlogspotCommentFeedDetector

  @Test
  def testShouldDetectBlogspotCommentFeedUrl(): Unit = assertTrue(detector.isValid(BLOGSPOT_COMMENT_URL, source))

  @Test
  def testShouldIgnoreBlogspotAtomFeed(): Unit = assertFalse(detector.isValid(BLOGSPOT_ATOM_URL, source))

}